keywords/MobileRestClient.py

```diff
@@ -1761,6 +1761,34 @@ def add_design_doc(self, url, db, name, doc):
             resp_obj = resp.json()
             return resp_obj["id"]
 
+    def get_design_doc_rev(self, url, db, name):
+        """
+        Keyword that gets a Design Doc revision
+        """
+        resp = self._session.get("{}/{}/_design/{}".format(url, db, name))
+        log_r(resp)
+        resp.raise_for_status()
+
+        # Only return a response if adding to the listener
+        # Sync Gateway does not return a response
+        if self.get_server_type(url) == ServerType.listener:
+            resp_obj = resp.json()
+            return resp_obj["_rev"]
+
+    def update_design_doc(self, url, db, name, doc, rev):
+        """
+        Keyword that updates a Design Doc to the database
+        """
+        resp = self._session.put("{}/{}/_design/{}?rev={}".format(url, db, name, rev), data=doc)
+        log_r(resp)
+        resp.raise_for_status()
+
+        # Only return a response if adding to the listener
+        # Sync Gateway does not return a response
+        if self.get_server_type(url) == ServerType.listener:
```

Comment:
SG does not return anything:
» http put <SG_IP>:4985/sg_db1/_design/dd\?rev=1 content_view='{"map":"function(doc, meta){if(doc.content){emit(doc._id,doc._rev);}}"}}}'
HTTP/1.1 201 Created
Content-Length: 0
Content-Type: text/plain; charset=utf-8
Date: Wed, 03 May 2017 22:27:44 GMT
Server: Couchbase Sync Gateway/1.4.1